src/types.ts

```diff
@@ -106,3 +106,10 @@ export interface Viewport {
     maxZoom: number
     minZoom: number
 }
+
+export type NonEmpty<A> = [A, ...A[]]
+
+export function nonEmpty<A>(arr: A[]): NonEmpty<A> | undefined {
```

Comment:
Denne høyrer kanskje meir heime under ei form for `utils` enn `types`? Resten her er vel TypeScript-typer.